app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,7 @@
+class FoodsController < ApplicationController
+  def search
+    @query = params[:q]
```

Comment:
No need for this to be set as an instance variable here. The view only needs access to `@hits` and `@display_foods`.